src/main/scala/za/co/absa/commons/lang/extensions/StringExtension.scala

```diff
@@ -0,0 +1,246 @@
+     * @param escape      the special character to escape the expected behavior within string
+     * @return true if anything is found, false otherwise
+     */
+    def hasUnquoted(charsToFind: Set[Char], quoteChars: Set[Char], escape: Char = '\\'): Boolean = {
+      findFirstUnquoted(charsToFind, quoteChars, escape).nonEmpty
+    }
+
+    /**
+     * Counts the occurrences of the chars to find. The occurrence is not considered if the character is part of a
+     * sequence within a pair of quote characters specified by quoteChars param.
+     * Escape character in front of a quote character will cancel its "quoting" function.
+     * Escape character in front of a searched-for character will not result in positive identification of a find
+     * Double escape character is considered to be escape character itself, without its special meaning
+     * The escape character can be part of the charsToFind set or quoteChars set and the function will work as
+     * expected (e.g. double escape character being recognized as a searched-for character or quote character), but it
+     * cannot be both - that will fire an exception.
+     *
+     * @param charsToFind set of characters to look for
+     * @param quoteChars  set of characters that are considered as quotes, everything within two (same) quote characters
+     *                    is not considered
+     * @param escape      the special character to escape the expected behavior within string
+     * @return map where charsToFind are the keys and values are the respective number of occurrences
+     */
+    def countUnquoted(charsToFind: Set[Char], quoteChars: Set[Char], escape: Char = '\\'): Map[Char, Int] = {
+      checkInputsOverlap(charsToFind, quoteChars, escape: Char)
+      val resultInit: Map[Char, Int] = charsToFind.map((_, 0)).toMap
+      val examineInit: (Option[Char], Boolean) = (Option.empty, false)
+      val (result, _) = string.foldLeft((resultInit, examineInit))((acc, char) => {
+        val (resultAcc, (charToExitQuotes, escaped)) = acc
+        val examineResult = examineChar(Option(char), charsToFind, quoteChars, escape, charToExitQuotes, escaped)
+        examineResult.map((resultAcc, _)) //no hit, propagate the examineResult
+          .getOrElse(resultAcc ++ Map(char -> (resultAcc(char) + 1)), examineInit)
+      })
+      result
+    }
+
+    /**
+     * Joins two strings with / while stripping single existing trailing/leading "/" in between:
+     * {{{
+     * "abc" / "123" -> "abc/123"
+     * "abc/" / "123" -> "abc/123"
+     * "abc" / "/123" -> "abc/123"
+     * "abc/" / "/123" -> "abc/123",
+     * but:
+     * "file:///" / "path" -> "file:///path",
+     * }}}
+     *
+     * @param another the second string we are appending after the `/` separator
+     * @return this/another (this has stripped trailing / if present, another has leading / stripped if present)
+     */
+    def /(another: String): String = { // scalastyle:ignore method.name
```

Comment:
Is this something worth to have in _commons_?

Reply:
Good question. To me, this is the only natural usage of `/` operator on strings, as I'm not aware of any natural or at least somewhat commonly used [field](https://en.wikipedia.org/wiki/Field_(mathematics)) structure on strings. So in my opinion it is worth to have it here.

Reply:
Good spotted. I also think it's not quite intuitive. By analogy with the standard `*` operator on `String` I would expect `/` would do the opposite, e.i.:
```scala
"abc" * 3 == "abcabcabc"
"abcabcabc" / 3 == "abc"
```
(not telling it's something practically useful thought, but such semantics would just be less of surprise to me I guess)

In your case `/` is used for a specific case of string concatenation, something like `+` but smarter :)
I would say you can accomplish almost the same by simply using `Path.of(a, b).toString` without involving any additional commons function call. That would cover all but the last case (the one with `file:///`)

```scala
     val myPath = Path.of("abc/", "/123").toString // self explanatory
     // versus
     val myPath = "abc/" / "/123" // kinda magic
```
The last `file:///` case, I'm not sure how exactly you use it in Enceladus, but I would guess the string `file:///` is likely only exist as a literal in the code. If so than a simple string concat would do the same thing.

Also, for string concatenation it's recommended to prefer string interpolation over `+` or other ways.

Reply:
> Is this something worth to have in _commons_?

Of course, I am author-biased, but I find it quite useful for cases where user's input on path is expected and concatenation of paths (specifically prefixes, relative path and file names) is happening. Without this, it is tiresome to otherwise check trailing/leading slashes when doing e.g.
```scala
rootPath / relativePath / fileName
```
where the `rootPath` for example may be given both as `/path/to/here` or `/path/to/here/`, and analogically for the rest.

With @wajda's examples of
```scala
"abc" * 3 == "abcabcabc"
"abcabcabc" / 3 == "abc"
```
I agree that this is (`*`) / could be (`/`) the expected behavior, but this is specialized to `Int` params. With a `String` param, the expectation, I believe can be different.

My vote on this matter would be, since you apparently find it not generic enough to stay in generic `StringOps` implicit, which I can see the motivation for, to move this into a different implicit wrapper, perhaps named `PathJoin`, so this can still be (re)used easily, but within a proper context.

Reply:
> move this into a different implicit wrapper
that would be fine IMO

Although why is `Path.of(rootPath, relativePath, fileName)` not enough?

Reply:
As far as I know, Path.of is platform-specific (different separator on Windows and rest of the world), wheras `/` would be platform-agnostic.

I agree to move it to a separate implicit class, but I'm gonna name it `StringConcatenationOps`, as the `/` functionality can be used also outside paths context. What do you think? 

Then the convention for implicit extensions in the library would be as I have written in the description of the PR + if some functionality is not general enough for a given type put it in `TypeExtension.Type<Usecase>Ops`.